cogni-sem/src/test/java/zone/cogni/sem/jena/SparqlUtilsTest.java

```diff
@@ -0,0 +1,58 @@
+package zone.cogni.sem.jena;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import zone.cogni.libs.sparqlservice.impl.JenaModelSparqlService;
+
+public class SparqlUtilsTest {
+  private static final String URI = "https://fedlex.data.admin.ch/eli/oc/2023/423/legal-analysis";
+  private static final String GRAPH_URI = URI + "/graph";
+  private final JenaModelSparqlService jenaModelSparqlService = new JenaModelSparqlService();
+
+  @Test
+  public void test() {
+    testEscape("A\ncomment"); // test newline
+    testEscape("A\\comment"); // test backslash
+    testEscape("A\bcomment"); // test backspace
+    testEscape("I'mAComment"); // test single quote
+    testEscape("A\fcomment"); // test form feed
+    testEscape("A\tcomment");  // test tab
+    testEscape("A\rcomment"); // test carriage return
+    testEscape("com\"ment"); // test double quotes
+  }
+
+  private void testEscape(String value) {
+    initializeSource(value);
+    confirmResult(value);
+  }
+
+  private void initializeSource(String value) {
+    jenaModelSparqlService.executeUpdateQuery(
+      "PREFIX jolux: <http://data.legilux.public.lu/resource/ontology/jolux#> " +
+      "INSERT DATA { " +
+      "  GRAPH <" + GRAPH_URI + "> { " +
+      "    <" + URI + "> a jolux:LegalAnalysis. " +
+      "    <" + URI + "> jolux:impactFromLegalResourceComment \"" + SparqlUtils.escapeString(value) + "\"" +
```

Comment:
@christinaspachiou did you validate the fact that you define a literal with single quote also?
would be good to do the tests for both cases (single quote and double quote)

Reply:
yes I had tested, but I added that here too